src/uu/ps/src/ps.rs

```diff
@@ -0,0 +1,195 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+mod collector;
+mod mapping;
+mod parser;
+mod picker;
+
+use clap::crate_version;
+use clap::{Arg, ArgAction, ArgMatches, Command};
+use mapping::{collect_code_mapping, default_codes, default_mapping};
+use parser::{parser, OptionalKeyValue};
+use prettytable::{format::consts::FORMAT_CLEAN, Row, Table};
+use std::{cell::RefCell, rc::Rc};
+use uu_pgrep::process::walk_process;
+use uucore::{
+    error::{UError, UResult, USimpleError},
+    format_usage, help_about, help_usage,
+};
+
+const ABOUT: &str = help_about!("ps.md");
+const USAGE: &str = help_usage!("ps.md");
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    let snapshot = walk_process()
+        .map(|it| Rc::new(RefCell::new(it)))
+        .collect::<Vec<_>>();
+    let mut proc_infos = Vec::new();
+
+    proc_infos.extend(collector::basic_collector(&snapshot));
+    proc_infos.extend(collector::process_collector(&matches, &snapshot));
+    proc_infos.extend(collector::session_collector(&matches, &snapshot));
```

Comment:
I think this approach is not correct. For example, if I provide `-A`, I get duplicate processes because I get the processes from `basic_collector` plus those from `process_collector` (which returns all processes).